src/prompt.c

```diff
@@ -446,6 +446,7 @@ readline_init(void)
 
 static void sigint_absorb_handler(int sig) {
 	signal(SIGINT, SIG_DFL);
+	redraw_display(true);
```

Comment:
Good point. How about moving `redraw_display` out of `sigint_absorb_handler` and to the end of `read_prompt`? Based on your comment, I came to the conclusion that it probably shouldn't be in `sigint_absorb_handler` at all.